packages/frontend-v2/pages/signup.tsx

```diff
@@ -37,81 +70,97 @@ const Signup: NextPage = () => {
       redirectToOnboardingOrHome(user, router);
     } catch (err) {
       const error = err as AxiosError;
-      if (error.response?.status === 401)
-        setApiError("Invalid credentials, please try again.");
-      else setApiError(error.message);
-      logger.error(error);
+      if (error.response?.status === 401) toast.error("Invalid Credentials!");
+      else toast.error("Something went wrong!");
     }
   };
+
   return (
-    <form onSubmit={handleSubmit(onSubmitHandler)}>
-      <h2>Sign Up</h2>
-      <br />
+    <FormFormat onSubmit={handleSubmit(onSubmitHandler)}>
+      <SignUpFormTopInput
+        errors={errors}
+        password={password}
+        register={register}
+      />
+      <SignUpFormButton isSubmitting={isSubmitting} router={router} />
+    </FormFormat>
+  );
+};
 
-      {apiError && (
-        <Text
-          pt="5%"
-          fontSize={{ desktop: "3xl", laptop: "2xl", tablet: "xl" }}
-          color="red.300"
-        >
-          {apiError}
-        </Text>
-      )}
+const SignUpFormTopInput: React.FC<SignUpFormTopProps> = ({
+  errors,
+  register,
+  password,
+}) => (
+  <HeaderAndInput>
+    <Text
+      fontSize="3xl"
+      color="primary.red.main"
+      as="b"
+      mb="xl"
+      textAlign="center"
+    >
+      Welcome!
+    </Text>
 
-      <FormControl isInvalid={errors.email != null}>
-        <Input
-          id="email"
-          placeholder="Example@email.com"
-          {...register("email", {
-            required: "Email is required",
-            pattern: {
-              value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
-              message: "Invalid email address",
-            },
-          })}
-        />
-        <FormErrorMessage>{errors.email?.message}</FormErrorMessage>
-      </FormControl>
+    <InputGroup>
+      <StringInput
+        id="email"
+        placeholder="Email"
+        error={errors.email}
+        type="email"
+        {...register("email", {
+          required: "Email is required",
+          pattern: {
+            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
```

Comment:
It is probably better to use a schema validation library like `yup` or `zod` to check for correct input type here since using regex is prone to errors and not sustainable.

Reply:
We discussed using this earlier in the logic portion but decided against since it didn't make sense to complicate the frontend validation